packages/data/src/namespace-store/index.js

```diff
@@ -169,68 +220,43 @@ function mapResolvers( resolvers, selectors, fulfillment, store ) {
 					return;
 				}
 
-				if ( fulfillment.hasStarted( selectorName, args ) ) {
+				const { metadata } = store.__unstableOriginalGetState();
+				if ( metadataSelectors.hasStartedResolution( metadata, selectorName, args ) ) {
 					return;
 				}
 
-				fulfillment.start( selectorName, args );
-				await fulfillment.fulfill( selectorName, ...args );
-				fulfillment.finish( selectorName, args );
+				store.dispatch( metadataActions.startResolution( selectorName, args ) );
+				await fulfillResolver( store, mappedResolvers, selectorName, ...args );
```

Comment:
I seem to be running into an issue where this doesn't actually wait for the fulfillment to complete before marking the resolution as finished:

![Screen Shot 2019-03-29 at 11 58 03 AM](https://user-images.githubusercontent.com/1779930/55246062-cc328100-521a-11e9-9fc9-aa3eecf8fe28.png)

_(Note in last three actions, I'd have expected "RECEIVE_ITEMS" to have occurred prior to "FINISH_RESOLUTION")_

I tried to consider how to reproduce this outside my branch, but since we [wait for most all data to be available](https://github.com/WordPress/gutenberg/blob/39171ca106b288b6ae1af06563a013a52b84ed2b/packages/edit-post/src/editor.js#L77) before rendering anything, it's hard to recreate even when adding a filter like `add_filter( 'block_editor_preload_paths', '__return_empty_array' );`. Which has me wondering if it could in-fact be an issue which had already existed prior to this change, and only became evident when performing resolution for data which hadn't been initially requested in the display of the editor (in my case, [waiting to resolve `getPostType( 'wp_block' )`](https://github.com/WordPress/gutenberg/blob/39171ca106b288b6ae1af06563a013a52b84ed2b/packages/editor/src/store/actions.js#L309-L313) when saving a reusable block).

I'll try to clean up my local working branch to push later this afternoon, but raising now in case anything comes to mind.

Reply:
I found the issue. I'll have a pull request shortly.